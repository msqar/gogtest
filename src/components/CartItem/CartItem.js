import React from 'react';
import { Col, Image, Row } from 'react-bootstrap';

class CartItem extends React.PureComponent {

    handleProductRemoval = () => {
        const { item, onRemove } = this.props;
        if (onRemove) onRemove(item.id);
    }

    render() {
        const { item } = this.props;
        return (
            <div className="CartItem">
                <Row>
                    <Col lg={4}>
                        <Image className="CartItem-thumbnail" src={item.thumbnail} />
                    </Col>
                    <Col lg={6} className="u-paddingAn">
                        <div className="CartItem-body">
                            <span className="CartItem-title">{item.title}</span>
                            <button onClick={this.handleProductRemoval} className="CartItem-removeButton Button--textOnly">Remove</button>
                        </div>
                    </Col>
                    <Col lg={2} className="u-paddingLn">
                        <div className="CartItem-priceContainer">
                            <span className="CartItem-priceTag">$ {item.price}</span>
                        </div>
                    </Col>
                </Row>
            </div>
        )
    }
}

export default CartItem;