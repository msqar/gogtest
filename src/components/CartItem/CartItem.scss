.CartItem {

}

.CartItem-thumbnail {
    
}

.Cartitem-body {

}

.CartItem-title {

}

.CartItem-priceTag {

}