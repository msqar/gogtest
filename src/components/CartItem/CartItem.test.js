import React from 'react';
import renderer from 'react-test-renderer';
import CartItem from './CartItem';

const mockedItem = {
    thumbnail: '/images/game_thumbnail_1.png',
    price: 9.99,
    title: 'Some Game Title'
};

it('renders Cart correctly', () => {
    const tree = renderer.create(<CartItem item={mockedItem} />).toJSON();
    expect(tree).toMatchSnapshot();
});