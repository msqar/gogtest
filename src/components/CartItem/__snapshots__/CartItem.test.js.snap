// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Cart correctly 1`] = `
<div
  className="CartItem"
>
  <div
    className="CartItem-thumbnailContainer"
  >
    <img
      className="CartItem-thumbnail"
      src="/images/game_thumbnail_1.png"
    />
  </div>
  <div
    className="CartItem-bodyWrap"
  >
    <div
      className="CartItem-body"
    >
      <span
        className="CartItem-title"
      >
        Some Game Title
      </span>
      <button
        className="CartItem-removeButton Button--textOnly"
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
    <div
      className="CartItem-priceContainer"
    >
      <span
        className="CartItem-priceTag"
      >
        $ 
        9.99
      </span>
    </div>
  </div>
</div>
`;
